packages/fxa-auth-server/test/local/routes/subscriptions.js

```diff
@@ -0,0 +1,409 @@
+
+      assert.equal(customs.check.callCount, 1, 'calls customs.check');
+
+      assert.equal(subhub.listPlans.callCount, 1);
+      assert.deepEqual(
+        subhub.createSubscription.args,
+        [
+          [ UID, PAYMENT_TOKEN_VALID, PLANS[0].plan_id ]
+        ]
+      );
+      assert.equal(db.createAccountSubscription.callCount, 1);
+
+      const createArgs = db.createAccountSubscription.args[0][0];
+      assert.deepEqual(
+        createArgs,
+        {
+          uid: UID,
+          subscriptionId: SUBSCRIPTION_ID_1,
+          productName: PLANS[0].product_id,
+          createdAt: createArgs.createdAt
+        }
+      );
+      assert.deepEqual(
+        res,
+        { subscriptionId: SUBSCRIPTION_ID_1 }
+      );
+    });
+
+    it('should correctly handle payment backend failure on listing plans', async () => {
+      subhub.listPlans = sinon.spy(async () => {
+        throw error.backendServiceFailure();
+      });
+      try {
+        await runTest(
+          '/subscriptions/active',
+          {
+            ...requestOptions,
+            method: 'POST',
+            payload: {
+              ...requestOptions.payload,
+              plan_id: PLANS[0].plan_id,
+              token: PAYMENT_TOKEN_VALID,
+            },
+          }
+        );
+        assert.fail();
+      } catch (err) {
+        assert.deepEqual(err.errno, error.ERRNO.BACKEND_SERVICE_FAILURE);
+        assert.equal(subhub.createSubscription.callCount, 0);
+        assert.equal(db.createAccountSubscription.callCount, 0);
+      }
+    });
+
+    it('should correctly handle payment backend failure on create', async () => {
+      subhub.createSubscription = sinon.spy(async () => {
+        throw error.backendServiceFailure();
+      });
+      try {
+        await runTest(
+          '/subscriptions/active',
+          {
+            ...requestOptions,
+            method: 'POST',
+            payload: {
+              ...requestOptions.payload,
+              plan_id: PLANS[0].plan_id,
+              token: PAYMENT_TOKEN_VALID,
+            },
+          }
+        );
+        assert.fail();
+      } catch (err) {
+        assert.deepEqual(err.errno, error.ERRNO.BACKEND_SERVICE_FAILURE);
+        assert.equal(db.createAccountSubscription.callCount, 0);
+      }
+    });
+
+    it('should correctly handle an unknown plan', async () => {
+      try {
+        await runTest(
+          '/subscriptions/active',
+          {
+            ...requestOptions,
+            method: 'POST',
+            payload: {
+              ...requestOptions.payload,
+              plan_id: 'thisisnotaplan',
+              token: PAYMENT_TOKEN_VALID,
+            },
+          }
+        );
+        assert.fail();
+      } catch (err) {
+        assert.deepEqual(err.errno, error.ERRNO.UNKNOWN_SUBSCRIPTION_PLAN);
+        assert.equal(db.createAccountSubscription.callCount, 0);
+      }
+    });
+
+    it('should correctly handle payment token rejection', async () => {
```

Comment:
Can users create multiple subscriptions to the same service? Or would that fail?

Reply:
Hmm, that's a good question - I think that's something we might need to enforce at the subhub level